tests/test_cli.py

```diff
@@ -176,10 +185,176 @@ def test_correct_call_DownloadManager_with_relative_time(
+            cli_copy.main()
+        self.assertEqual(
+            [call(dst='/dst_folder/', src='/tmp/testing_file.test'),
+             call(dst='/dst_folder/', src='/new_loc_add')],
+            mock_copy.call_args_list)
+        for dataset in Dataset.objects.all():
+            dataset.dataseturi_set.filter(
+                uri='/new_loc_add', dataset=dataset, service=LOCAL_FILE_SERVICE).delete()
+
+    @mock.patch('os.path.isfile', return_value=True)
+    def test_correct_place_of_symlink_after_creation_of_it(self, mock_isfile):
+        """ symlink must be placed at the address that is specified from the input arguments """
+        sys.argv = [
+            "",
+            '-b', "2018-04-01",
+            '-e', "2018-04-09",
+            '-l',
+            '-d', "/dst_folder/"
+        ]
+        for dataset in Dataset.objects.all():
+            dataset.dataseturi_set.get_or_create(
+                uri='/new_loc_add', dataset=dataset, service=LOCAL_FILE_SERVICE)
+        with mock.patch('os.symlink') as mock_symlink:
+            cli_copy.main()
+        self.assertEqual(
+            [call(dst='/dst_folder/testing_file.test', src='/tmp/testing_file.test'),
+             call(dst='/dst_folder/new_loc_add', src='/new_loc_add')],
+            mock_symlink.call_args_list)
+        for dataset in Dataset.objects.all():
+            dataset.dataseturi_set.filter(
+                uri='/new_loc_add', dataset=dataset, service=LOCAL_FILE_SERVICE).delete()
+
+    @mock.patch('os.symlink')
+    @mock.patch('os.path.isfile', return_value=True)
+    def test_correct_content_of_flag_file(self, mock_isfile, mock_link):
+        """ flag file should contain this 'type: test_type' information """
+        sys.argv = [
+            "",
+            '-b', "2018-04-01",
+            '-e', "2018-04-09",
+            '-f', '-l',
+            '-t', 'test_type'
+        ]
+        temp_directory = tempfile.TemporaryDirectory()
+        sys.argv.append('-d')
+        sys.argv.append(temp_directory.name)
+        cli_copy.main()
+        with open(os.path.join(temp_directory.name + '/testing_file.test.flag'), 'r') as fd:
+            self.assertEqual(fd.read(), (
+                "type: test_type\nurl: https://scihub.copernicus.eu/apihub/odata/v1/Produ"
+                "cts('6127111d-c9bd-4689-bab5-412dd39e1e81')/$value\nurl: https://scihu"
+                "b.copernicus.eu/the_second_fakeurl\n"
+            )
+        )
+
+    def test_copying_accomplishment_and_consistency_for_symlink_of_file_copying(self):
+        """ A symlink of actual sample file should be copied to the destination folder """
+        sys.argv = [
+            "",
+            '-b', "2018-04-01",
+            '-e', "2018-04-09",
+            '-f', '-l'
+        ]
+        temp_directory_dst = tempfile.TemporaryDirectory()
+        sys.argv.append('-d')
+        sys.argv.append(temp_directory_dst.name)
+        Path("/tmp/testing_file.test").touch()
```

Comment:
Shouldn't `"/tmp/testing_file.test"` be `tempfile.TemporaryFile`?

Reply:
Discussed beforehand! Here is the start of discussion: https://github.com/nansencenter/django-geo-spaas-processing/pull/24#discussion_r526108351

Reply:
Keep it simple! We have something else in our mind when we wrote this test! No need to engage ourselves with TemporaryFile.

On the other hand, TemporaryFile will be destroyed as soon as it is closed as its doc revealed https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryFile!  which is not our case here!

Reply:
You can use `tempfile.TemporaryFile` as a context manager (`with temfile.TemporaryFile...`) to keep it open as long as you need.

Reply:
When we use a temporary folder which contains that file, then why should we bother ourselves two code with a context manager? the subject is not that file! the subject is something else! Now the test is written in a `without indentation` manner and it is simpler to understand than using a context manager.

Reply:
The problem is that you may not have the `/tmp` folder, or not have writing permissions. `tempfile.TemporaryFile` creates temporary files in a safe and reliable manner. Please don't reinvent the wheel and hardcode paths. We should use the available Python functionality. It is the same as using `os.path.join(a,b)` instead of `a + '/' + b`. The latter seems simpler but it is not correct.

Reply:
These are two difference subjects! I did not reinvent the wheel! I have suggested to use `a.rstrip(os.path.sep) + os.path.sep + b.lstrip(os.path.sep)` not `a + '/' + b`. It is obvious that `a + '/' + b` is naive. I never suggested `a + '/' + b`.

`/tmp` folder is the suggestion of @aperrin66 and as far as I know, the linux machine always have that folder based on the https://en.wikipedia.org/wiki/Filesystem_Hierarchy_Standard.

It seems I can't convince you two not to deal with a context manager!!!!
You are just making the tests complicated unnecessarily!
I will do it with context manager only because of your insist.

Reply:
@aperrin66 @akorosov 
I have just found that the address comes from the database and we need that file to be there based on that address which is installed to the database based on the fixtures! So it is impossible to user a tempfile as a context manager for this case. tempfile has a varying address but what is inside the fixtures is a constant address! Even `tempfile.NamedTemporaryFile` in addtion to `tempfile.TemporaryFile` does not provide a specific name for us to create a file with that name. THESE ARE ALL TEMPORARY.

Perhaps SOMETIMES it is better that you reconsider your decision(s) of your suggestions when I insist on something!

 Please review the code carefully, there are some time spend on writing the code, so it needs to spend time for reviewing it as well.

Reply:
If you 'have just found that' in the code that you wrote yourself, how do you expect me to find that? Please write good code that doesn't need to look up for hardcoded values in several places. Then it will be much easier to review (which is not an easy job either).

The problem with using `/tmp` is that it can potentially be unachievable. That's why `TemporaryFile` was developed and added to Python standard library - to provide safe and reliable access to temporary directories and files. As we have discussed many times, it is better to use standard approaches than develop something new. Usage of `os.path.join` is just an example. It is not related to usage of `/tmp`. It exemplifies that we can code either `a + '/' + 'b'` or `os.path.join(a, b)`. Although it seems more complicated, but the latter approach is correct because it uses the tools specially developed for this operation. The same here: we should not use '/tmp' because it is not a universal approach, it is not standard. It will fail, e.g. on Windows, or if you don't have access to '/tmp' or, maybe in other circumstances. Therefore we should not use it.

If we cannot use '/tmp' there are several ways to avoid using it. In some cases using `TemporaryFile` will help to avoid using '/tmp'. But apparently not in this case, because you hardcode this path in the fixtures. But it doesn't mean that we still should use '/tmp'. It means that we should look for other solutions. And that's what I expect from productive discussion of my review - attempt to identify the problem and attempt to find a solution to this problem. Not just a refusal of fixing problems.

If we cannot use `TemporaryFile` we can use something else. What's eventually needed in this test is to make a symbolic link in the call to `cli_copy.main()` and check that no errors are raised, right? And there are many ways to do that without using `/tmp`. Without even using `TemporaryDirectory`, or without actually creating a symbolic link.
Shall I suggest some?